hudi-utilities/src/test/java/org/apache/hudi/utilities/deser/TestKafkaAvroSchemaDeserializer.java

```diff
@@ -0,0 +1,141 @@
+  }
+
+  private Schema createUserSchema() {
+    String userSchema = "{\"namespace\": \"example.avro\", \"type\": \"record\", "
+        + "\"name\": \"User\","
+        + "\"fields\": [{\"name\": \"name\", \"type\": \"string\"}]}";
+    Schema.Parser parser = new Schema.Parser();
+    Schema schema = parser.parse(userSchema);
+    return schema;
+  }
+
+  private IndexedRecord createUserRecord() {
+    Schema schema = createUserSchema();
+    GenericRecord avroRecord = new GenericData.Record(schema);
+    avroRecord.put("name", "testUser");
+    return avroRecord;
+  }
+
+  private Schema createExtendUserSchema() {
+    String userSchema = "{\"namespace\": \"example.avro\", \"type\": \"record\", "
+        + "\"name\": \"User\","
+        + "\"fields\": [{\"name\": \"name\", \"type\": \"string\"}, "
+        + "{\"name\": \"age\", \"type\": [\"null\", \"int\"], \"default\": null}]}";
+    Schema.Parser parser = new Schema.Parser();
+    Schema schema = parser.parse(userSchema);
+    return schema;
+  }
+
+  private IndexedRecord createExtendUserRecord() {
+    Schema schema = createExtendUserSchema();
+    GenericRecord avroRecord = new GenericData.Record(schema);
+    avroRecord.put("name", "testUser");
+    avroRecord.put("age", 30);
+    return avroRecord;
+  }
+
+  /**
+   * Tests {@link KafkaAvroSchemaDeserializer#deserialize(boolean, String, Boolean, byte[], Schema)}.
+   */
+  @Test
+  public void testKafkaAvroSchemaDeserializer() {
+    byte[] bytesOrigRecord;
+    IndexedRecord avroRecord = createUserRecord();
+    SchemaTestProvider.schemaToReturn.set(origSchema);
+    KafkaAvroSchemaDeserializer avroDeserializer = new KafkaAvroSchemaDeserializer(schemaRegistry, new HashMap(defaultConfig));
+    avroDeserializer.configure(new HashMap(defaultConfig), false);
+    bytesOrigRecord = avroSerializer.serialize(topic, avroRecord);
+    // record is serialized in orig schema and deserialized using same schema.
+    assertEquals(avroRecord, avroDeserializer.deserialize(false, topic, false, bytesOrigRecord, origSchema));
+
+    IndexedRecord avroRecordWithAllField = createExtendUserRecord();
+    byte[] bytesExtendedRecord = avroSerializer.serialize(topic, avroRecordWithAllField);
+
+    SchemaTestProvider.schemaToReturn.set(evolSchema);
+    avroDeserializer = new KafkaAvroSchemaDeserializer(schemaRegistry, new HashMap(defaultConfig));
+    avroDeserializer.configure(new HashMap(defaultConfig), false);
+    // record is serialized w/ evolved schema, and deserialized w/ evolved schema
+    IndexedRecord avroRecordWithAllFieldActual = (IndexedRecord) avroDeserializer.deserialize(false, topic, false, bytesExtendedRecord, evolSchema);
+    assertEquals(avroRecordWithAllField, avroRecordWithAllFieldActual);
+    assertEquals(avroRecordWithAllFieldActual.getSchema(), evolSchema);
+
+    // read old record w/ evolved schema.
+    IndexedRecord actualRec = (IndexedRecord) avroDeserializer.deserialize(false, topic, false, bytesOrigRecord, origSchema);
```

Comment:
@vburenin @nsivabalan I guess the comment is somewhat misleading here. The comment says "read old record with evolved schema", however we are passing `origSchema` only as the last argument in deserialize method. Am I missing something here?

Reply:
My bad, got it now.